amt/client.py

```diff
@@ -81,19 +81,26 @@ def __init__(self, address, password, username='admin', protocol='http'):
         self.password = password
 
     def post(self, payload, ns=None):
-        resp = requests.post(self.uri,
-                             headers={'content-type':
-                                      'application/soap+xml;charset=UTF-8'},
-                             auth=HTTPDigestAuth(self.username, self.password),
-                             data=payload)
-        resp.raise_for_status()
-        if ns:
-            rv = _return_value(resp.content, ns)
-            if rv == 0:
-                return 0
-            print(pp_xml(resp.content))
-        else:
-            return 0
+        try:
```

Comment:
I'd rather not do the try here. If this is consumed as a library, it makes sense for these exceptions to flow up to the next layer so that they can handle these errors however they like. 

Instead I would put the exception handling and error messages printed to the user here - https://github.com/sdague/amt/blob/d3e5d742a12ae68b2c60298c30ccdad9c92e5bb2/bin/amtctrl#L115